### PR TITLE
Fix for go vet warning on ldaputil/client.go

### PR DESCRIPTION
**Pull request by abstractj**
Fixes: https://github.com/openshift/origin/issues/12393

@enj I believe this fixes the issue.

### CODE REVIEW

File: pkg/auth/ldaputil/client.go [hunk 1]
```diff
@@ -66,7 +66,7 @@ func (l *ldapClientConfig) Connect() (ldap.Client, error) {
 	// Ensure tlsConfig specifies the server we're connecting to
 	if tlsConfig != nil && !tlsConfig.InsecureSkipVerify && len(tlsConfig.ServerName) == 0 {
 		// Add to a copy of the tlsConfig to avoid mutating the original
-		c := *tlsConfig
+		c := CloneTLSConfig(tlsConfig)
```
Comment (enj) [inline pkg/auth/ldaputil/client.go]: Use `k8s.io/apimachinery/pkg/util/net.CloneTLSConfig`.

File: pkg/auth/ldaputil/client.go [hunk 2]
```diff
@@ -105,6 +105,29 @@ func (l *ldapClientConfig) Connect() (ldap.Client, error) {
 	}
 }
 
+// TODO: replace with tls.Config#Clone when we move to go1.8
+func CloneTLSConfig(c *tls.Config) *tls.Config {
```
Comment (enj) [inline pkg/auth/ldaputil/client.go]: Remove.

**Comment by abstractj** [comment]
@enj That's SO true! Changed it and rebased.

**Comment by enj** [comment]
@abstractj please open an issue to use Clone once we switch to go1.8.

[test]

[testextended][extended:ldap_groups]

**Comment by enj** [comment]
```
github.com/openshift/origin/pkg/auth/ldaputil
pkg/auth/ldaputil/client.go:76: cannot use &c (type **tls.Config) as type *tls.Config in assignment
```

**Comment by abstractj** [comment]
[test]

[testextended][extended:ldap_groups]

**Comment by stevekuznetsov** [comment]
@abstractj you'll want to remove the entry from the [`go vet` blacklist](https://github.com/openshift/origin/blob/master/hack/verify-govet.sh#L7) as well

**Comment by enj** [comment]
@stevekuznetsov this took me forever to find in the logs:

```
./pkg/auth/ldaputil/client.go
!!! gofmt needs to be run on the following files: 
Try running 'gofmt -s -d [path]'
Or autocorrect with 'hack/verify-gofmt.sh | xargs -n 1 gofmt -s -w'
```

@abstractj perhaps you could update that code to print `[FAILURE]` for easier grepping?

**Comment by enj** [comment]
Flake #14043

**Comment by abstractj** [comment]
@enj @stevekuznetsov thanks, I just updated this PR like you suggested.

@enj regards to print [FAILURE] for easier grepping, I ca do that. Just not sure what exactly you mean.

**Comment by stevekuznetsov** [comment]
Could apply a patch like:

```patch
From 134c646acb3565e3feb33d5cb9854346f4aa74ea Mon Sep 17 00:00:00 2001
From: Steve Kuznetsov <skuznets@redhat.com>
Date: Mon, 26 Jun 2017 11:29:11 -0700
Subject: [PATCH] Update output logging in hack/verify-gofmt.sh

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>
---
 hack/verify-gofmt.sh | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/hack/verify-gofmt.sh b/hack/verify-gofmt.sh
index 3d00509..92ffe73 100755
--- a/hack/verify-gofmt.sh
+++ b/hack/verify-gofmt.sh
@@ -5,9 +5,8 @@ os::golang::verify_go_version
 
 bad_files=$(os::util::list_go_src_files | xargs gofmt -s -l)
 if [[ -n "${bad_files}" ]]; then
-	echo "!!! gofmt needs to be run on the following files: " >&2
+	os::log::error "!!! gofmt needs to be run on the following files: "
 	echo "${bad_files}"
-	echo "Try running 'gofmt -s -d [path]'" >&2
-	echo "Or autocorrect with 'hack/verify-gofmt.sh | xargs -n 1 gofmt -s -w'" >&2
-	exit 1
+	os::log::fatal "Try running 'gofmt -s -d [path]'
+Or autocorrect with 'hack/verify-gofmt.sh | xargs -n 1 gofmt -s -w'"
 fi
-- 
2.9.4

```

**Comment by abstractj** [comment]
@stevekuznetsov thanks Steve, patch applied.

**Comment by stevekuznetsov** [comment]
[merge]

**Comment by smarterclayton** [comment]
[test]

**Comment by enj** [comment]
@abstractj congrats \o/

**Comment by abstractj** [comment]
@enj thanks my friend!

**Pull request merged**